webapp/publisher/snaps/views.py

```diff
@@ -1345,42 +1351,90 @@ def get_snap_builds(snap_name):
         "snap_id": details["snap_id"],
         "snap_name": details["snap_name"],
         "snap_title": details["title"],
+        "snap_builds_enabled": False,
+        "snap_builds": [],
     }
 
-    lp_snaps = launchpad.snaps.findByStoreName(
-        owner="https://api.launchpad.net/devel/~build.snapcraft.io",
-        store_name=details["snap_name"],
-    )
-
     github = GitHubAPI(flask.session.get("github_auth_secret"))
     context["github_user"] = github.get_user()
 
     if context["github_user"]:
         context["github_repositories"] = github.get_user_repositories()
 
-    context["snap_builds_enabled"] = len(lp_snaps) >= 1
-    if context["snap_builds_enabled"]:
-        context["snap_builds"] = []
+    # Get built snap in launchpad with this store name
+    lp_snap = launchpad.get_snap_by_store_name(details["snap_name"])
+
+    if lp_snap:
+        # Git repository without GitHub hostname
+        context["github_repository"] = lp_snap["git_repository_url"][19:]
+
         bsi_url = flask.current_app.config["BSI_URL"]
 
-        for build in lp_snaps[0].builds:
-            link = build_link(bsi_url, lp_snaps[0], build)
+        builds = launchpad.get_collection_entries(
+            lp_snap["builds_collection_link"]
+        )
+
+        context["snap_builds_enabled"] = bool(builds)
+
+        for build in builds:
+            link = build_link(bsi_url, lp_snap, build)
             status = map_build_and_upload_states(
-                build.buildstate, build.store_upload_status
+                build["buildstate"], build["store_upload_status"]
             )
 
             context["snap_builds"].append(
                 {
-                    "id": build.self_link.split("/")[-1],
-                    "arch_tag": build.arch_tag,
-                    "datebuilt": build.datebuilt,
-                    "duration": build.duration,
+                    "id": build["self_link"].split("/")[-1],
+                    "arch_tag": build["arch_tag"],
+                    "datebuilt": build["datebuilt"],
+                    "duration": build["duration"],
                     "link": link,
-                    "logs": build.build_log_url,
-                    "revision_id": build.revision_id,
+                    "logs": build["build_log_url"],
+                    "revision_id": build["revision_id"],
                     "status": status,
-                    "title": build.title,
+                    "title": build["title"],
                 }
             )
 
     return flask.render_template("publisher/builds.html", **context)
+
+
+@publisher_snaps.route("/<snap_name>/builds", methods=["POST"])
+@login_required
+def get_snap_builds_post(snap_name):
```

Comment:
Shouldn't this be called `post_snap_builds`?

Reply:
:flushed: You are right!